pandas/core/indexes/multi.py

```diff
@@ -235,16 +235,28 @@ def set_levels(self, levels, level=None, inplace=False,
                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
                    names=[u'foo', u'bar'])
         """
-        if level is not None and not is_list_like(level):
-            if not is_list_like(levels):
-                raise TypeError("Levels must be list-like")
-            if is_list_like(levels[0]):
-                raise TypeError("Levels must be list-like")
+
+        level_list_like = level is None or is_list_like(level)
+        levels_list_like = (is_list_like(levels) and
+                            all(is_list_like(x) for x in levels) and
+                            levels != [] )
+
+        if level_list_like:
+            # level is a list-like object of scalars
+            levels_error = ("`levels` and `level` are incompatible. "
+                            "When `level` is list-like, `levels` must be a "
+                            "list-like object containing list-like objects")
+            if not levels_list_like:
+                raise TypeError(levels_error)
+        elif not level_list_like:
+            # level is a scalar
+            levels_error = ("`levels` and `level` are incompatible. "
+                            "When `level` is a scalar, `levels` must be a "
+                            " list-like object of scalars.")
+            if levels_list_like:
```

Comment:
I *think* this will behave incorrectly for setting `index.set_index(levels=[], level=0)`. That *should* work, just setting that level's levels to `[]`. In other words, this test should pass

```python
def test_thing(self):
    index = pd.MultiIndex(levels=[['a', 'b'], [1, 2]], labels=[[], []])
    result = index.set_levels([], level=0)
    expected = pd.MultiIndex(levels=[[], [1, 2]], labels=[[], []])
    tm.assert_index_equal(result, expected)
```

lmk if that makes sense. I may be incorrect.